### DISCUSSION TITLE
Gateway Example question? https://github.com/hyperium/hyper/blob/master/examples/gateway.rs

### DISCUSSION THREAD

**Discussion by ZNackasha** [hyperium/hyper#3774] [Q&A] [answered]
in the gateway example. we create a new connection for each request that comes in from the listener.

![image](https://github.com/user-attachments/assets/14d6640c-d209-4542-a29d-c4b446e01000)

is there a way to do this without the reconnection on each request? I don't want to redo the handshake constantly. I would love it if every message simply gets forwarded. 

my use case is to check cookies and see if the user has valid permissions and if they do I would forward them along.

**Comment by seanmonstar** [comment, marked as answer]
You can look at `hyper_util::client::legacy::Client` for a connection pool.
